src/app/api/users/[userId]/route.ts

```diff
@@ -0,0 +1,58 @@
+import dbConnect from "../../../../lib/mongodb/mongodb";
+import { UserModel } from "../../../../lib/schemas/user.schema";
+import { getServerSession } from "next-auth";
+import { NextRequest, NextResponse } from "next/server";
+import { hash } from "bcrypt";
+import authOptions from "../../auth/auth-options";
+
+
+ export async function GET(req: NextRequest, {params}:{ params: { userId: string }}){
+    
+    const _id = params.userId;
+
+    const session = await getServerSession(authOptions);
+    if (session?.user?._id !==_id) {
+        return NextResponse.json({ error: "Unauthorized" }, { status: 401 });
+    }
```

Comment:
Here you can move this check to the PATCH route because that's where the resource is being updated.
For this route, you can simply use `getServerSession` to make sure the requesting user just has a session. The idea is that as long as a user is authenticated, the front should be able to get any user by their id.